lib/mi_bridges/driver/base_page.rb

```diff
@@ -68,6 +74,27 @@ def next
 
       attr_reader :snap_application, :logger
 
+      def log(description, *text)
+        if logger.present?
+          logger.tagged(self.class.to_s) do
+            logger.debug("#{description.upcase}: #{text.join(', ')}")
+          end
+        end
+      end
+
+      def skip_instance_limitation?
+        self.class.instance_methods(false).include?(:skip_instance_limitation)
+      end
+
+      def check_instance_limitation
```

Comment:
maybe this could be renamed to something like `check_infinite_loop` so the meaning is clearer?